code/modules/spells/spell.dm

```diff
@@ -118,10 +137,10 @@ GLOBAL_LIST_INIT(spells, typesof(/obj/effect/proc_holder/spell)) //needed for th
 	var/critfailchance = 0
 	var/centcom_cancast = 1 //Whether or not the spell should be allowed on z2
 
-	var/action_icon = 'icons/mob/actions/actions_spells.dmi'
-	var/action_icon_state = "spell_default"
-	var/action_background_icon_state = "bg_spell"
-	var/datum/action/spell_action/action
+	action_icon = 'icons/mob/actions/actions_spells.dmi'
+	action_icon_state = "spell_default"
+	action_background_icon_state = "bg_spell"
+	datum/action/spell_action/spell/action
```

Comment:
again

Reply:
this line is fucky but it turns out to be exactly what i want since i'd have to make a whole new var, and set it manually at some point (since action is set externally, not on Initialize), if i don't redefine the type

Reply:
it's not fucky at all, the subtype can totally override the parent var type as long as they're the same overall type